app/middleware/authenticator.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require './app/models/user'
+
+def logged_in?
+  !session[:user_id].nil?
+end
+
+# Define the condition
+set(:auth) do |_condition|
```

Comment:
make this a before filter and call it in a block with all methods that require authentication before accessing